addons/api/mirage/factories/account.js

```diff
@@ -11,32 +11,48 @@ import {
   TYPE_AUTH_METHOD_PASSWORD,
   TYPE_AUTH_METHOD_OIDC,
   TYPE_AUTH_METHOD_LDAP,
+  TYPES_AUTH_METHOD,
 } from 'api/models/auth-method';
 
 export default factory.extend({
-  authorized_actions: () =>
-    permissions.authorizedActionsFor('account') || [
-      'no-op',
-      'read',
-      'update',
-      'delete',
-      'set-password',
-    ],
+  authorized_actions() {
+    let authorizedActions = ['no-op', 'read', 'update', 'delete'];
+    if (this.type === TYPE_AUTH_METHOD_PASSWORD)
+      authorizedActions.push('set-password');
+    return permissions.authorizedActionsFor('account') || authorizedActions;
+  },
 
   id: () => generateId('acct_'),
 
+  type: (i) => TYPES_AUTH_METHOD[i % TYPES_AUTH_METHOD.length],
+
   attributes() {
-    switch (this.type) {
-      case TYPE_AUTH_METHOD_PASSWORD:
-      case TYPE_AUTH_METHOD_LDAP:
-        return { login_name: faker.random.words() };
-      case TYPE_AUTH_METHOD_OIDC:
-        return {
-          issuer: faker.internet.ip(),
-          subject: 'sub',
-          email: faker.internet.email(),
-          full_name: faker.random.words(),
-        };
+    if (this.type === TYPE_AUTH_METHOD_PASSWORD) {
+      return { login_name: faker.internet.userName() };
+    }
+
+    if (this.type === TYPE_AUTH_METHOD_LDAP) {
+      const groupsAmount = faker.datatype.number({ min: 1, max: 3 });
```

Comment:
I'm curious, why did you change this from a `switch` to `if` statements? Is it because of the variable declaration in this block and the linter complained?

If so you can get around that by explicitly making a switch case statement have a block, i.e. wrap it in braces

Reply:
Yup, exactly the reason. Thank you! Updated!